control-plane/Makefile

```diff
@@ -43,8 +43,15 @@ ci.dev-docker:
 	$(CI_DEV_DOCKER_WORKDIR) -f $(CURDIR)/Dockerfile
 	@echo "Pushed dev image to: $(CI_DEV_DOCKER_NAMESPACE)/$(CI_DEV_DOCKER_IMAGE_NAME):$(GIT_COMMIT)"
 ifeq ($(CIRCLE_BRANCH), main)
-	@docker tag $(CI_DEV_DOCKER_NAMESPACE)/$(CI_DEV_DOCKER_IMAGE_NAME):$(GIT_COMMIT) $(CI_DEV_DOCKER_NAMESPACE)/$(CI_DEV_DOCKER_IMAGE_NAME):latest
-	@docker push $(CI_DEV_DOCKER_NAMESPACE)/$(CI_DEV_DOCKER_IMAGE_NAME):latest
+	@docker buildx build -t '$(CI_DEV_DOCKER_NAMESPACE)/$(CI_DEV_DOCKER_IMAGE_NAME):latest' \
```

Comment:
Does this also need the `docker buildx create --use &&` piece to use the builder?

Reply:
we're doing it in circle as a previous step so I think it'll be sufficient. It passed on my test run here: https://app.circleci.com/pipelines/github/hashicorp/consul-k8s/6644/workflows/62728521-110c-42df-878b-65bba814ee7e/jobs/55300.

Reply:
Sounds good!